src/hotspot/cpu/riscv/riscv.ad

```diff
@@ -1238,17 +1238,24 @@ bool needs_acquiring_load_reserved(const Node *n)
 
 int MachCallStaticJavaNode::ret_addr_offset()
 {
-  // jal
-  return 1 * NativeInstruction::instruction_size;
+  if (UseTrampolines) {
+    return 1 * NativeInstruction::instruction_size; // jal
+  }
+  return 3 * NativeInstruction::instruction_size; // auipc + ld + jalr
```

Comment:
Question: As we will only patch the address in the stub, do we still need the handling in compute_padding (`CallStaticJavaDirectNode::compute_padding` & `CallDynamicJavaDirectNode::compute_padding`) when `UseTrampolines` is false?

Reply:
No, not that I know of. But we need additional fixes, some asserts needs to tweaked and PostCallNop expects aligned calls (as we want to patch nop -> trap not crossing pages) so we instead need padding after.

I think we can either have:
`c.nop, auipc, ld, jalr, nop (cmodx to trap), nop`
`auipc, ld, jalr, c.nop, nop (cmodx to trap), nop`

It seems to me keeping what we have is just simplest.

What you reckon?

Reply:
Ah, I didn't notice the PostCallNop at the end. I agree it will be safer to keep the current shape.